src/lucky_record/where.cr

```diff
@@ -115,29 +123,29 @@ module LuckyRecord::Where
 
   class In < SqlClause
     def operator
-      "= ANY"
+      "IN"
     end
 
     def negated : NotIn
       NotIn.new(@column, @value)
     end
 
     def prepare(prepared_statement_placeholder : String)
-      "#{column} #{operator} (#{prepared_statement_placeholder})"
+      "#{column} #{operator} (SELECT * FROM unnest(#{prepared_statement_placeholder}#{type_cast}))"
```

Comment:
I finally found a way to use an `IN` statement with an array, but the array needs to be casted to a specific type. So I added a type_cast method and restricted `.in()` calls to `Array(Int32)` and `Array(String)`.